packages/contracts/deploy/001-ChugSplashManager.ts

```diff
@@ -0,0 +1,93 @@
+/* Imports: External */
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+
+const deployFn: DeployFunction = async (hre) => {
+  const { deployer } = await hre.getNamedAccounts()
+
+  const proxyUpdaterAddress = (await hre.deployments.get('ProxyUpdater'))
+    .address
+
+  // Deploy the ChugSplashManager implementation contract.
+  const { deploy: deployManager } = await hre.deployments.deterministic(
+    'ChugSplashManager',
+    {
+      salt: hre.ethers.utils.solidityKeccak256(
+        ['string'],
+        ['ChugSplashManager']
+      ),
+      from: deployer,
+      log: true,
+    }
+  )
+  const { address: managerAddress, abi: managerAbi } = await deployManager()
+
+  // Get the ChugSplashRegistry's proxy address that *will* be deployed using Create2. We'll need
+  // this address when we initialize the ChugSplashManagerProxy.
+  // First, get the artifact of the Proxy.
+  const registryProxyArtifact = await hre.artifacts.readArtifact(
+    '@eth-optimism/contracts-bedrock/contracts/universal/Proxy.sol:Proxy'
+  )
+  // ABI encode the proxy's constructor argument.
+  const encodedConstructorArg = hre.ethers.utils.defaultAbiCoder.encode(
+    ['address'],
+    [deployer]
+  )
+  // Get the proxy's init code, which is the proxy's bytecode with the encoded constructor argument
+  // appended.
+  const initCode = `${
+    registryProxyArtifact.bytecode
+  }${encodedConstructorArg.slice(2)}`
+  // Get the Proxy address generated by Create2.
+  const registryProxyAddress = hre.ethers.utils.getCreate2Address(
+    '0x4e59b44847b379578588920ca78fbf26c0b4956c', // Deterministic deployment proxy address
+    hre.ethers.utils.solidityKeccak256(['string'], ['Proxy']), // Salt
+    hre.ethers.utils.keccak256(initCode) // Init code hash
+  )
+
+  // Deploy and initialize the ChugSplashManagerProxy:
+  // First, ABI encode the `initialize` function call.
+  const encodedInitializeFunction = new hre.ethers.utils.Interface(
+    managerAbi
+  ).encodeFunctionData('initialize', [
+    registryProxyAddress,
+    'Upgradeable Manager',
+    deployer,
+    proxyUpdaterAddress,
+  ])
+  // Then, deploy and initialize the proxy.
+  const { deploy: deployProxy } = await hre.deployments.deterministic(
+    'ChugSplashManagerProxy',
+    {
+      salt: hre.ethers.utils.solidityKeccak256(
+        ['string'],
+        ['ChugSplashManagerProxy']
+      ),
+      from: deployer,
+      args: [
+        registryProxyAddress,
+        managerAddress,
+        deployer,
+        encodedInitializeFunction,
+      ],
+      log: true,
+    }
+  )
+  await deployProxy()
+
+  // Initialize the implementation contract. This won't make any difference during testing, since
```

Comment:
Can remove this if we add that constructor to the implementation

Reply:
Added a call to the `initialize` function in the constructors. I left the state variables without the `immutable` keyword unless you've got a solution to [this](https://github.com/smartcontracts/chugsplash/pull/54#discussion_r960151869) though